pkg/policy/validate.go

```diff
@@ -132,90 +133,103 @@ func Validate(policy *kyverno.ClusterPolicy, client *dclient.Client, mock bool,
 	for i, rule := range policy.Spec.Rules {
 		//check for forward slash
 		if err := validateJSONPatchPathForForwardSlash(rule.Mutation.PatchesJSON6902); err != nil {
-			return fmt.Errorf("path must begin with a forward slash: spec.rules[%d]: %s", i, err)
+			return nil, fmt.Errorf("path must begin with a forward slash: spec.rules[%d]: %s", i, err)
 		}
 
 		if jsonPatchOnPod(rule) {
 			log.Log.V(1).Info("pods managed by workload controllers cannot be mutated using policies, use the auto-gen feature or write policies that match pod controllers")
+			return &v1beta1.AdmissionResponse{
+				Allowed:  true,
+				Warnings: []string{"pods managed by workload controllers cannot be mutated using policies, use the auto-gen feature or write policies that match pod controllers"},
+			}, nil
 		}
+
+		if policy.HasAutoGenAnnotation() {
+			log.Log.V(1).Info("pod controllers excluded by autogen annotation requires additional work of adding preconditions to exclude the controller as part of the policy")
+			return &v1beta1.AdmissionResponse{
+				Allowed:  true,
+				Warnings: []string{"pod controllers excluded by autogen annotation requires additional work of adding preconditions to exclude the controller as part of the policy"},
```

Comment:
Suggested:

"Pod controllers excluded from autogen require adding of preconditions to also exclude the desired controller(s)."

Reply:
Made the recommended changes 👍🏽